src/components/Modals/NewCourse/NewCourse.vue

```diff
@@ -277,38 +282,35 @@ export default Vue.extend({
       });
       /* execute a function presses a key on the keyboard: */
       inp.addEventListener('keydown', e => {
-        let x = document.getElementById(`${inp.id}autocomplete-list`);
-        if (x) x = x.getElementsByTagName('div');
+        const x = document.getElementById(`${inp.id}autocomplete-list`) as HTMLDivElement;
+        const xList: readonly HTMLDivElement[] = x ? [...x.getElementsByTagName('div')] : [];
         if (e.keyCode === 40) {
           /* If the arrow DOWN key is pressed,
             increase the currentFocus variable: */
           currentFocus += 1;
           /* and and make the current item more visible: */
-          addActive(x);
+          addActive(xList);
         } else if (e.keyCode === 38) {
           // up
           /* If the arrow UP key is pressed,
             decrease the currentFocus variable: */
           currentFocus -= 1;
           /* and and make the current item more visible: */
-          addActive(x);
+          addActive(xList);
         } else if (e.keyCode === 13) {
           /* If the ENTER key is pressed, prevent the form from being submitted, */
           e.preventDefault();
           if (currentFocus > -1) {
             /* and simulate a click on the "active" item: */
-            if (x) x[currentFocus].click();
+            if (xList) xList[currentFocus].click();
```

Comment:
Observed some weird behavior. Left is PR, right is master. 

My actions: Start typing in class, use keyboard (down arrow key) to move to the first option, press `Enter` key (so that the first option is the text of the search bar), press `Enter` key again, I get this console message that points to this line. I can't seem to replicate this on master, so maybe it has to do with how `xList` is defined in this cleanup?

https://user-images.githubusercontent.com/54298311/106843685-952e2180-6674-11eb-8cad-d0f0ed17335d.mov

Reply:
Actually it's probably not worth fixing it in this diff. I already put up the follow up diff that doesn't have this issue: https://github.com/cornell-dti/course-plan/pull/259, and it has a much better implementation. Once that's approved I can merge it together.